src/pantry.js

```diff
@@ -0,0 +1,27 @@
+// let rawIngredientsData = ingredientsData !== undefined ? ingredientsData : require('../data/ingredients')
+
+let rawIngredientsData = require('../data/ingredients')
+
+class Pantry {
+  constructor(userIngredients) {
+    this.ingredients = this.generateFullIngredientList(userIngredients);
+  }
+
+  generateFullIngredientList(partialIngredients) {
+      return partialIngredients.map(ingredient => {
+
+      let matchedIngredient = rawIngredientsData.find(rawIngredient =>{
+          return ingredient.ingredient === rawIngredient.id
+        })
+
+        return {name:matchedIngredient.name,
+                id: matchedIngredient.id,
+                estimatedCostInCents: matchedIngredient.estimatedCostInCents,
+                amount: ingredient.amount
+              };
+    })
```

Comment:
this might be a good spot for the "calculate cost of ingredients still needed for each recipe" method.